### DISCUSSION TITLE
[Feature Request] add ScrollBar Component for better looks

### DISCUSSION THREAD

**Discussion by Reverier-Xu** [heroui-inc/heroui#1241] [Feature requests]
### Is your feature request related to a problem? Please describe.

the default scrollbar on some browser is ugly:

![image](https://user-images.githubusercontent.com/41937333/210711209-e8b25b00-74eb-4112-bdc6-8f53dcc21ce4.png)

Using CSS to customize the scrollbar style is one way to solve this, but the default scrollbars that take up the width of the content may cause some visual problems.

### Describe the solution you'd like

I'd like something like this:

![image](https://user-images.githubusercontent.com/41937333/210711993-4b32ded6-611a-4273-be1c-5ddd613f391d.png)

Scrollbars can automatically hide when not scrolling and float above the content when needed, without taking up a fixed width.

### Describe alternatives you've considered

just like other component libraries do: https://www.naiveui.com/en-US/dark/components/scrollbar

**Comment by rnwelsh** [comment]
The native scrollbars have pseudoclasses for styling, `::-webkit-scrollbar`, `::-webkit-scrollbar-thumb`, etc.  They're all non-standard though and a bit limited with what you can do. 

As for the hiding effect, I just use `overflow: overlay`and set a transparent color on all scrollbar elements, and then do something like `.scroll-parent:hover::-webkit-scrollbar-thumb {background-color: #069420;}` 
It's not perfect but it works

**Comment by Reverier-Xu** [comment]
> The native scrollbars have pseudoclasses for styling, `::-webkit-scrollbar`, `::-webkit-scrollbar-thumb`, etc. They're all non-standard though and a bit limited with what you can do.
> 
> As for the hiding effect, I just use `overflow: overlay`and set a transparent color on all scrollbar elements, and then do something like `.scroll-parent:hover::-webkit-scrollbar-thumb {background-color: #069420;}` It's not perfect but it works

Thank you! But the `overflow: overlay` is deprecated and not supported by all the browsers, such as Firefox. I think we still need something to do this.

At present , I am using [OverlayScrollbars](https://kingsora.github.io/OverlayScrollbars/) , it works well, but it is still a little troublesome when integrated with `textarea`.
